projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -0,0 +1,165 @@
+const debug = require( '../../debug' );
+
+/* global GitHub, WebhookPayloadIssue */
+
+/**
+ * Search for a previous comment from this task in our issue.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - Issue number.
+ * @returns {Promise<number>} Promise resolving to a number.
+ */
+async function getListComment( octokit, owner, repo, number ) {
+	let commentID = 0;
+
+	debug( `gather-support-references: Looking for a previous comment from this task in our issue.` );
+
+	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
+		owner,
+		repo,
+		issue_number: +number,
+	} ) ) {
+		response.data.map( comment => {
+			if (
+				comment.user.login === 'github-actions[bot]' &&
+				comment.body.includes( '**Support References**' )
+			) {
+				commentID = comment.id;
+			}
+		} );
+	}
+
+	return commentID;
+}
+
+/**
+ * Scan the contents of the issue as well as all its comments, get all support references, and add them to an array of references.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - Issue number.
+ * @returns {Promise<Array>} Promise resolving to an array.
+ */
+async function getIssueReferences( octokit, owner, repo, number ) {
+	const ticketReferences = [];
+	const supportIds = [];
+	const referencesRegexP = /[0-9]*-(?:chat|hc|zen|zd)/gim;
+
+	debug( `gather-support-references: Getting references from comments.` );
+	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} ) ) {
+		response.data.map( comment => {
+			ticketReferences.push( ...comment.body.matchAll( referencesRegexP ) );
```

Comment:
Might be worth having the bot ignore its own comment. Otherwise if someone adds a wrong task, then deletes it, the bot would preserve it since it would still find it in its own comment.

Reply:
That's a good point. Done in 59903a1a33e06f7150ddafab548eedaf8bab6146